netcheck/netcheck.go

```diff
@@ -454,11 +458,66 @@ func (c *Client) GetReport(ctx context.Context) (*Report, error) {
 		}
 	}
 
-	// TODO: if UDP is blocked, try to measure TCP connect times
-	// to DERP nodes instead? So UDP-blocked users still get a
-	// decent DERP node, rather than being randomly assigned to
-	// the other side of the planet? Or try ICMP? (likely also
-	// blocked?)
+	// Try HTTPS if UDP latency check failed for every stun server
+	// TODO: It does not show that UDP latency check failed
+	client := http.DefaultClient
+	client.Timeout = time.Second * 5
+	var rmu sync.Mutex
+	checkLatencyHTTPS := func(server string) error {
+		c.logf("netcheck: %s UDP latency check failed, try HTTPS", server)
+		// Maybe there is a better way to get https url
+		sp := strings.Split(server, ":")
+		if len(sp) == 0 {
+			return errors.New("netcheck: checkLatencyHTTPS: cannot get domain from stun server")
+		}
+		u := fmt.Sprintf("https://%s/derp", sp[0])
```

Comment:
let's use a new explicit path for this, like /latency-check

Reply:
Do we need to add a new endpoint on the server?